biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -124,12 +152,39 @@ subroutine create_cohort(currentSite, patchptr, pft, nn, hite, dbh, bleaf, bfine
     new_cohort%canopy_layer = clayer
     new_cohort%canopy_layer_yesterday = real(clayer, r8)
     new_cohort%laimemory    = laimemory
-    new_cohort%bdead        = bdead
-    new_cohort%bstore       = bstore
-    new_cohort%bl           = bleaf
-    new_cohort%br           = bfineroot
-    new_cohort%bsw          = bsap
-    new_cohort%ode_opt_step = 1.0e6_r8   ! Initialize the integrator step size as super-huge
+
+
+    ! Initialize the Plant allocative Reactive Transport (PaRT) module
+    ! Choose from one of the extensible hypotheses (EH)
+    ! -----------------------------------------------------------------------------------
+
+    call InitPRTCohort(new_cohort)
+
+    ! The initialization allocates memory, but the boundary and initial
+    ! contitions must be set. All new cohorts go through create_cohort()
+    ! so this should be the only place this is called.  Alternatively
+    ! cohorts can be copied and fused, but special routines handle that.
+    ! -----------------------------------------------------------------------------------
+
+    select case(hlm_parteh_model)
+    case (prt_carbon_allom_hyp)
+
```

Comment:
I'm assuming that the existence of this one case is because there's only one (c-only) option now, but i guess that the later cases would have their own sets of statements here? Just making sure I'm getting this right...

Reply:
The C+N+P hypotheses, when it is added, will add a block to the select case.
> select case( hlm_parteh_mode)
> case (prt_carbon_allom_hyp)
> stuff
> case (prt_cnp_allom_hyp) 
> stuff
> end